lxd/main_netcat.go

```diff
@@ -21,27 +24,83 @@ func cmdNetcat(args []string) error {
 		return fmt.Errorf("Bad arguments %q", args)
 	}
 
+	// The socket name on LXD instances prior to netcat log file is:
+	// 	/<some-tmp-path/lxd_rsync_<random-suffix>
+	// and on LXD instances with netcat log file:
+	// 	/<some-tmp-path/lxd_<ctName>_rsync_<random-suffix>
+	// Detect the latter case and create a netcat log
+	var logFile *os.File
+	var err error
+	ctName := ""
+	logPath := ""
+	namestart := strings.LastIndex(args[1], "/lxd_")
```

Comment:
I tend to stay away from regexp unless absolutely needed, they rarely improve code readability and don't have the best of security track records (though I'm not aware of any regexp compiler CVEs for Go). In a compiled and heavily optimized language like Go, they're also unlikely to give you the performance boost that they would in scripting languages.